GirafIntegrationTest/tests/test_account_controller.py

```diff
@@ -0,0 +1,355 @@
+        response = post(f'{BASE_URL}v1/Account/login', json=data).json()
+        self.assertTrue(response['success'])
+        self.assertEqual(response['errorKey'], 'NoError')
+        self.assertIsNotNone(response['data'])
+        graatand_token = response['data']
+
+    @order
+    def test_account_get_graatand_id(self):
+        """
+        Testing getting Graatand's id
+
+        Endpoint: GET:/v1/User
+        """
+        global graatand_id
+        response = get(f'{BASE_URL}v1/User', headers=auth(graatand_token)).json()
+        self.assertTrue(response['success'])
+        self.assertEqual(response['errorKey'], 'NoError')
+        self.assertIsNotNone(response['data']['id'])
+        graatand_id = response['data']['id']
+
+    @order
+    def test_account_register_grundenberger(self):
+        """
+        Testing registering Grundenberger
+
+        Endpoint: POST:/v1/Account/register
+        """
+        global grundenberger_username
+        grundenberger_username = f'Grundenberger{time.time()}'
+        data = {'username': grundenberger_username, 'password': 'password', 'role': 'Citizen', 'departmentId': 1}
+        response = post(f'{BASE_URL}v1/Account/register', json=data, headers=auth(graatand_token)).json()
+        self.assertTrue(response['success'])
+        self.assertEqual(response['errorKey'], 'NoError')
+
+    @order
+    def test_account_login_as_grundenberger(self):
+        """
+        Testing logging in as Grundenberger
+
+        Endpoint: POST:/v1/Account/login
+        """
+        global grundenberger_token
+        data = {'username': grundenberger_username, 'password': 'password'}
+        response = post(f'{BASE_URL}v1/Account/login', json=data).json()
+        self.assertTrue(response['success'])
+        self.assertEqual(response['errorKey'], 'NoError')
+        self.assertIsNotNone(response['data'])
+        grundenberger_token = response['data']
+
+    @order
+    def test_account_get_grundenberger_id(self):
+        """
+        Testing getting Grundenberger's id
+
+        Endpoint: GET:/v1/User
+        """
+        global grundenberger_id
+        response = get(f'{BASE_URL}v1/User', headers=auth(grundenberger_token)).json()
+        self.assertTrue(response['success'])
+        self.assertEqual(response['errorKey'], 'NoError')
+        self.assertIsNotNone(response['data']['id'])
+        grundenberger_id = response['data']['id']
+
+    @order
+    def test_account_get_username_with_auth(self):
+        """
+        Testing getting username using authorization
+
+        Endpoint: GET:/v1/User
+        """
+        response = get(f'{BASE_URL}v1/User', headers=auth(graatand_token)).json()
+        self.assertTrue(response['success'])
+        self.assertEqual(response['errorKey'], 'NoError')
+        self.assertIsNotNone(response['data']['username'])
+        self.assertEqual(response['data']['username'], 'Graatand')
+
+    @order
+    def test_account_login_invalid_password(self):
```

Comment:
```suggestion
    def test_when_invalid_password_Should_fail(self):
```

Or something like that. The `_Should` is always my to-go naming on python tests, my point is that the test names could be more descriptive.